### PR TITLE
Allow passing linodeApitoken and region as secretRef

### PR DESCRIPTION
**Pull request by schinmai-akamai**
### General:

* [x] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [x] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

Adding support for passing the LinodeAPIToken and region via secret references instead of raw helm values. This prevents the disclosure of the token in values files when using gitops practices.

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [x] Are you addressing a single feature in this PR? 
1. [x] Are your commits atomic, addressing one change per commit?
1. [x] Are you following the conventions of the language? 
1. [x] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [x] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

File: deploy/chart/templates/ccm-linode.yaml [hunk 1]
```diff
@@ -7,3 +8,4 @@ stringData:
   apiToken: {{ required ".Values.apiToken required" .Values.apiToken }}
   region: {{ required ".Values.region required" .Values.region }}
 type: Opaque
+{{- end }}
```
Comment (komer3) [inline deploy/chart/templates/ccm-linode.yaml:11]: nit: New line here

File: deploy/chart/values.yaml [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
-# region [Required] - Must be a Linode region. (https://api.linode.com/v4/regions)
+# region [Required if secretRef is not set] - Must be a Linode region. (https://api.linode.com/v4/regions)
 region: ""
 
+# Set these values if your APIToken and region are already present in a k8s secret.
+# secretRef:
+#   name: "linode-ccm"
+#   apiTokenRef: "apiToken"
+#   regionRef: "region"
+
```
Comment (komer3) [inline deploy/chart/values.yaml:12]: Question: Are we passing in the values when we install the helm chart? If yes, why is this part commented?

Comment (schinmai-akamai) [reply deploy/chart/values.yaml:12]: Yes, these values can be passed in when installing. This is commented to keep backwards compatibility. Existing users of the chart can continue to pass in the token and region, new users can choose to use the secretRef. Didn't want to break usersa

**Comment by srust** [comment]
@schinmai-akamai can you add a unit test for this?

**Comment by schinmai-akamai** [comment]
Helm linting passes with the token and region set manually
```
$ helm lint deploy/chart --set apiToken="apiToken",region="us-east"      
==> Linting deploy/chart
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```
Helm linting passes with the token and region set via secret
```
$ helm lint deploy/chart --set secretRef.apiTokenRef="apiToken",secretRef.name="api",secretRef.regionRef="us-east"                 
==> Linting deploy/chart
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

Helm template works with the secret being created when the apiToken and region is manually set
```
$  helm template foo deploy/chart --set apiToken="apiToken",region="us-east" --debug
---
# Source: ccm-linode/templates/ccm-linode.yaml
apiVersion: v1
kind: Secret
metadata:
  name: ccm-linode
  namespace: kube-system
stringData:
  apiToken: apiToken
  region: us-east
type: Opaque
---
# Source: ccm-linode/templates/daemonset.yaml
          env:
            - name: LINODE_API_TOKEN
              valueFrom:
                secretKeyRef:
                  name: "ccm-linode"
                  key: "apiToken"
            - name: LINODE_REGION
              valueFrom:
                secretKeyRef:
                  name: "ccm-linode"
                  key: "region"
```

Helm template works with secretReference - no secret, and correct values being passed.

```
$ helm template foo deploy/chart --set secretRef.apiTokenRef="apiToken",secretRef.name="api",secretRef.regionRef="us-east"
# Source: ccm-linode/templates/daemonset.yaml
          env:
            - name: LINODE_API_TOKEN
              valueFrom:
                secretKeyRef:
                  name: api
                  key: apiToken
            - name: LINODE_REGION
              valueFrom:
                secretKeyRef:
                  name: api
                  key: us-east
```

**Review by srust — APPROVED** [review]

**Pull request merged**